content/articles/machine-learning-with-xgboost-and-scikit-learn/index.md

```diff
@@ -0,0 +1,402 @@
+The process of converting categorical values into a numeric values is called categorical encoding.
+
+### Getting started with categorical encoding
+
+Before we start, let's get all the columns with the `object` datatype.
+
+```python
+df.columns[df.dtypes == 'object']
+```
+
+The output is shown.
+
+```bash
+Index(['job', 'marital', 'education', 'default', 'housing', 'loan', 'contact',
+       'month', 'day_of_week', 'poutcome', 'y'],
+      dtype='object')
+```
+
+To convert all the columns into numeric values we use `get_dummies()` method.
+
+`get_dummies()` is a pandas method that is used to convert the categorical data into encoded numerical values which are in a machine-readable form.
+
+```python
+pd.get_dummies(df,df.columns[df.dtypes == 'object'])
+```
+
+This will output a new dataset with encoded numeric values as shown below.
+
+![Categorical encoding](/engineering-education/machine-learning-with-xgboost-and-scikit-learn/categorical-encoding.jpg)
+
+The output shows the first `17` columns with encoded numeric values.
+
+For a detailed understanding and practical guide on how `get_dummies()` works, read this [guide](https://pandas.pydata.org/docs/reference/api/pandas.get_dummies.html)
+
+Let's see if the columns datatypes have changed.
+
+```python
+df.dtypes
+```
+
+The output is as shown in the image below.
+
+![Converted data types](/engineering-education/machine-learning-with-xgboost-and-scikit-learn/new-data-types.jpg)
+
+This shows all the `object` columns were converted into `int`. We can now start building our model.
+
+In this section, we will build our model using a basic Scikit-learn algorithm. We then later improve the model performance using XGBoost.
+
+### Installing XGBoost
+
+Let`s install XGBoost, since we are using Google Colab, we install using the following command.
+
+```python
+!pip install xgboost
+```
+
+Let's import this package.
+
+```python
+import xgboost as xgb
+```
+
+Now that we have imported XGBoost, let's split our dataset into the testing set and training set.
+
+### Dataset splitting
+
+Let's import `train_test_split` which is used for data set splitting.
+
+```python
+from sklearn.model_selection import train_test_split
+```
+
+We then split our dataset into two sets: train set and test set. 80% of the dataset will be used as a train set and 20% will be used as a test set.
+
+```python
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)
+```
+
+To understand the power of XGBoost we need to compare it with another algorithm, we first use the decision tree classifier algorithm to build the model. After that, we then build the same model using XGBoost and compare the results to see if XGBoost has improved the model performance.
+
+We start with a decision tree classifier.
+
+### Decision tree classifier
+
+A decision tree classifier is a machine learning algorithm that is used to solve classification problems. It's imported from the Scikit-learn Python library.
```

Comment:
It would be great if you give an insight on how it works.

Using inbuilt scikit library is easier by abstracting the working, but understanding it's working is essential.